app/services/customers/eu_auto_taxes_service.rb

```diff
@@ -4,14 +4,18 @@
 
 module Customers
   class EuAutoTaxesService < BaseService
-    def initialize(customer:)
+    def initialize(customer:, new_record:, changed_attributes:)
       @customer = customer
       @organization_country_code = customer.organization.country
+      @new_record = new_record
+      @changed_attributes = changed_attributes
 
       super
     end
 
     def call
+      return nil unless should_apply_eu_taxes?
```

Comment:
Ideally, we want to avoid returning nil and always return `result` so the calling method can safely call `result.success?` when any service is called.

Reply:
`nil` was returned since this service originally didn't work with result as a response (instead only string was returned).
But yep, probably this is the good timing to change it so refactored it:)